### PR TITLE
[UT] add ut for cmdinit/kubeconfig.go

### PR DESCRIPTION
**Pull request by my-git9**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Add ut for cmdinit/kubeconfig.og

**Which issue(s) this PR fixes**:
Ref https://github.com/karmada-io/karmada/issues/2579

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

File: pkg/karmadactl/cmdinit/utils/kubeconfig_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+		wantErr bool
+	}{
+		{
+			name: "Test Write KubeConfig From Spec",
+			args: args{
+				serverURL:      "https://127.0.0.1:6443",
+				userName:       "admin",
+				clusterName:    "local",
+				kubeconfigPath: "/tmp",
+				kubeconfigName: "karmada-test.kubeconfig",
```
Comment (XiShanYongYe-Chang) [inline pkg/karmadactl/cmdinit/utils/kubeconfig_test.go:137]: Hi @my-git9, this test case cannot pass the test on the Windows operating system:
```log
kubeconfig_test.go:148: WriteKubeConfigFromSpec() error = open \tmp\karmada-test.kubeconfig: The system cannot find the path specified., wantErr false
```

Comment (RainbowMango) [reply pkg/karmadactl/cmdinit/utils/kubeconfig_test.go:137]: nice finding. 
Please send another PR for this.

Comment (my-git9) [reply pkg/karmadactl/cmdinit/utils/kubeconfig_test.go:137]: Sorry, could you give me some advices for the path?

Comment (XiShanYongYe-Chang) [reply pkg/karmadactl/cmdinit/utils/kubeconfig_test.go:137]: Is it OK to generate a temporary file in the current directory?

Comment (my-git9) [reply pkg/karmadactl/cmdinit/utils/kubeconfig_test.go:137]: OK, I will try it later

**Pull request merged**